builtin/common/misc_helpers.lua

```diff
@@ -400,6 +400,80 @@ function core.explode_scrollbar_event(evt)
 	return retval
 end
 
+--------------------------------------------------------------------------------
+function core.explode_key_event(evt)
+	if evt == nil then
+		return nil
```

Comment:
`return nil` or just `return`? The latter is cleaner IMO (has to be documented appropriately though).